examples/ppi_unsup.py

```diff
@@ -0,0 +1,164 @@
+import os.path as osp
+
+import torch
+import torch.nn.functional as F
+from sklearn.linear_model import SGDClassifier
+from sklearn.metrics import f1_score
+from sklearn.multioutput import MultiOutputClassifier
+from torch_cluster import random_walk
+
+from torch_geometric.datasets import PPI
+from torch_geometric.loader import DataLoader, NeighborSampler
+from torch_geometric.nn import SAGEConv
+
+EPS = 1e-15
+
+path = osp.join(osp.dirname(osp.realpath(__file__)), '..', 'data', 'PPI')
+train_dataset = PPI(path, split='train')
+val_dataset = PPI(path, split='val')
+test_dataset = PPI(path, split='test')
+train_loader = DataLoader(train_dataset, batch_size=1, shuffle=True)
+val_loader = DataLoader(val_dataset, batch_size=2, shuffle=False)
+test_loader = DataLoader(test_dataset, batch_size=2, shuffle=False)
+
+train_data_list = [data for data in train_dataset]
+
+
+class NeighborSampler(NeighborSampler):
+    def sample(self, batch):
+        batch = torch.tensor(batch)
+        row, col, _ = self.adj_t.coo()
+
+        # For each node in `batch`, we sample a direct neighbor (as positive
+        # example) and a random node (as negative example):
+        pos_batch = random_walk(row, col, batch, walk_length=1,
+                                coalesced=False)[:, 1]
+
+        neg_batch = torch.randint(0, self.adj_t.size(1), (batch.numel(), ),
+                                  dtype=torch.long)
+
+        batch = torch.cat([batch, pos_batch, neg_batch], dim=0)
+        return super().sample(batch)
+
+
+train_loader_list = []
+
+for curr_graph in train_data_list:
+    _train_loader = NeighborSampler(curr_graph.edge_index, sizes=[10, 10],
+                                    batch_size=256, shuffle=True,
+                                    num_nodes=curr_graph.num_nodes)
+    train_loader_list.append(_train_loader)
```

Comment:
I don't think we want to use a list of samplers here. IMO, it is a better idea to first batch all graphs together, and then apply `NeighborSampler` on the entire set of nodes:
```python
batch = Batch.from_data_list(train_dataset)
loader = NeighborSampler(batch)
```

Reply:
Thanks, the results are 40% better now that I'm batching all train graphs together.